resolwe_bio/processes/samtools/samtools_mpileup.py

```diff
@@ -0,0 +1,367 @@
+                required=False,
+            )
+            min_mapping_qual = IntegerField(
+                label="Minimum mapping quality for an alignment to be used",
+                description="Only count reads with mapping quality greater than or equal to value specific. [-q]",
+                required=False,
+            )
+            excl_flags = ListField(
+                StringField(),
+                label="Filter flags",
+                default=["UNMAP", "SECONDARY", "QCFAIL", "DUP"],
+                description="Filter flags: skip reads with mask bits set. "
+                "Press ENTER after each flag. [-ff]",
+            )
+            excl_output_ends = BooleanField(
+                label="Exclude output ends in sequence column",
+                description="Removes the “^” (with mapping quality) and “$” markup from the sequence column [--no-output-ends].",
+                default=False,
+            )
+            output_base_counts = BooleanField(
+                label="Output per base counts",
+                description="Output base counts at each position. "
+                "This is a custom implementation and not part of the original samtools mpileup. "
+                "It outputs a file with per base counts at each position.",
+                default=False,
+            )
+
+        advanced = GroupField(AdvancedOptions, label="Advanced options")
+
+    class Output:
+        """Output fields for SamtoolsMpileupSingle."""
+
+        pileup_report = FileField(label="Output pileup report")
+        base_counts = FileField(label="Output per base counts", required=False)
+        species = StringField(label="Species")
+        build = StringField(label="Genome build")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        name = Path(inputs.bam.output.bam.path).stem
+        result_fn = f"{name}_mpileup.txt"
+
+        input_options = []
+        column_names = [
+            "chr",
+            "pos",
+            "ref",
+            "depth",
+            "base_sequence",
+            "base_qualities",
+        ]
+
+        if inputs.fasta:
+            if inputs.fasta.output.species != inputs.bam.output.species:
+                self.error(
+                    "Input BAM file and FASTA file are of different species. "
+                    f"BAM file is from {inputs.bam.output.species}, "
+                    f"while FASTA file is from {inputs.fasta.output.species}."
+                )
+            if inputs.fasta.output.build != inputs.bam.output.build:
+                self.error(
+                    "Input BAM and FASTA files should have matching genome builds. "
+                    "BAM file has build "
+                    f"{inputs.bam.output.build}, while FASTA file has build "
+                    f"{inputs.fasta.output.build}."
+                )
+            input_options.extend(["-f", inputs.fasta.output.fasta.path])
+
+        if inputs.bedfile:
```

Comment:
Begin with the check that either `bedfile` or `positions` is provided. ReSDK does not disable input fields in the same way as the FE can, so it is possible to provide both at the same time.